File: Frontend/src/server-models/cqrs/users/create-user.request.ts
import { AccountType } from 'src/server-models/enums/account-type.enum';
import { Gender } from 'src/server-models/enums/gender.enum';
export class CreateUserRequest {
    accountType: AccountType;
    username: string;
    email: string;
    firstname: string;
    lastname: string;
    gender: Gender;
    coachId: string;

    constructor() {
    }
   
}

File: Backend/Core/Business/Backend.Business.Users/UsersRequests/CreateUser/CreateUserRequest.cs
using Backend.Domain.Entities.User;
using Backend.Domain.Enum;
using MediatR;
using System;

namespace Backend.Business.Users.UsersRequests.CreateUser
{
    public class CreateUserRequest : IRequest<ApplicationUser>
    {
        public AccountType AccountType { get; set; }

        public string Username { get; set; }
        public string Email { get; set; }

        public string FirstName { get; set; }
        public string LastName { get; set; }

        public Gender Gender { get; set; }

        public Guid CoachId { get; set; }
    }
}